tests/test_training_args.py

```diff
@@ -0,0 +1,113 @@
+from unittest import TestCase
+
+import pytest
+
+from setfit.training_args import TrainingArguments
+
+
+class TestTrainingArguments(TestCase):
+    def test_training_args_raises_error_with_wrong_warmup_proportion(self):
+        # warmup_proportion must not be > 1.0
+        with pytest.raises(ValueError):
+            TrainingArguments(warmup_proportion=1.1)
+
+        # warmup_proportion must not be < 0.0
+        with pytest.raises(ValueError):
+            TrainingArguments(warmup_proportion=-0.1)
+
+    def test_training_args_batch_sizes(self):
+        batch_size_A = 12
+        batch_size_B = 4
+        batch_size_C = 6
+
+        args = TrainingArguments(batch_size=batch_size_A)
+        assert args.batch_size == (batch_size_A, batch_size_A)
```

Comment:
If I'm not mistaken, using `self.assertEqual` is the idiomatic way to check these cases, e.g.

```suggestion
        self.assertEqual(args.batch_size, (batch_size_A, batch_size_A))
```

Reply:
For `unittest` it indeed is. I'm used to the `pytest` approach which involves `assert` calls instead. I'll update this for consistency: better to go all `unittest` than mixing and matching too much.